frontend/src/metabase/query_builder/components/filters/FilterPopover/FilterPopover.unit.spec.tsx

```diff
@@ -36,82 +37,70 @@ const QUERY = Question.create({
     ["field", PRODUCTS.TITLE, { "source-field": ORDERS.PRODUCT_ID }],
     "asdf",
   ]);
-
-const [RELATIVE_DAY_FILTER, NUMERIC_FILTER, STRING_CONTAINS_FILTER] =
+const [RELATIVE_DAY_FILTER, NUMERIC_FILTER, STRING_CONTAINS_FILTER]: Filter[] =
   QUERY.filters();
 
 const dummyFunction = jest.fn();
 
+const setup = ({
+  filter: filterElem,
+  query = QUERY,
+  onChange = dummyFunction,
+  onChangeFilter = dummyFunction,
+  showFieldPicker = true,
+}: {
+  filter: Filter;
+  query?: StructuredQuery;
+  onChange?: (filter: Filter) => void;
+  onChangeFilter?: (filter: Filter) => void;
+  showFieldPicker?: boolean;
+}) => {
+  const filter = new Filter(filterElem, null, query);
```

Comment:
I don't get this part. What's the point of recreating a new filter here?

Why can't we just pass along `filter`? e.g.
```ts
const setup = ({
  filter,
  onChange = dummyFunction,
  onChangeFilter = dummyFunction,
  showFieldPicker = true,
}: {
  filter: Filter;
  query?: StructuredQuery;
  onChange?: (filter: Filter) => void;
  onChangeFilter?: (filter: Filter) => void;
  showFieldPicker?: boolean;
}) => {
  renderWithProviders(
    <FilterPopover
      query={QUERY}
      filter={filter}
      onChange={onChange}
      onChangeFilter={onChangeFilter}
      showFieldPicker={showFieldPicker}
    />,
  );
};
```

All tests still pass that way.

Reply:
Ah that makes sense considering the typing of `Filter` as an extension of `Array`. I was following what the previous test cases were doing where it was creating a new filter every time, didn't realize that could be nuked.